monte_carlo.py

```diff
@@ -102,9 +103,47 @@ def sample_location(x_min, x_max, y_min, y_max, z_min, z_max):
 #       neutron until its final absorption, noting the distance between the
 #       starting and ending point, appending this value to the crow_distances
 #       list.
+
+'''
+ @brief     Function that generates a neutron and measures how 
+            far it travels before being absorbed.
+ @details   A neutron is created in the bounding box using
+            sample_location(). It moves a distance determined by
+            sample_distance(). It is then either absorbed or
+            scattered as determined by sample_interaction(). When
+            it is absorbed, its distance from its starting point
+            is appended to crow_distances.
+ @param	    sigma_t the total cross-section 
+ @param     sigma_s the scattering cross-section
+ @param     x_min the minimum x-coordinate in the bounding box
+ @param     x_max the maximum x-coordinate in the bounding box
+ @param     y_min the minimum y-coordinate in the bounding box
+ @param     y_max the maximum y-coordinate in the bounding box
+ @param     z_min the minimum z-coordinate in the bounding box
+ @param     z_max the maximum z-coordinate in the bounding box
+ @param     crow_distances a list containg the distances
+            from each neutron's starting point to the point that
+            it is absorbed.
+'''
 def transport_neutron(sigma_t, sigma_s, x_min, x_max, y_min, y_max, z_min, 
-        z_max, crow_distances):
-    raise NotImplementedError
+    z_max, crow_distances):
+    neutron_starting_point = list(sample_location(x_min, x_max, y_min, y_max, z_min, z_max))
+    neutron_position = [neutron_starting_point[i] for i in [0,1,2]] 
+    neutron_interaction = 0
+    while neutron_interaction != 1:
+        neutron_distance = sample_distance(sigma_t)
+        theta = sample_polar_angle()
+        phi = sample_azimuthal_angle()
+        neutron_position[0] += neutron_distance*sin(phi)*cos(theta)
+        neutron_position[1] += neutron_distance*sin(phi)*sin(theta)
+        neutron_position[2] += neutron_distance*cos(phi)
+        neutron_interaction = sample_interaction(float(sigma_t), float(sigma_s))
+    neutron_distance_vector = [0, 0, 0]
+    for i in [0,1,2]:
+        neutron_distance_vector[i] = neutron_position[i] - neutron_starting_point[i]
+    crow_distance = sqrt(neutron_distance_vector[0]**2+neutron_distance_vector[1]**2+neutron_distance_vector[2]**2)
```

Comment:
Enforce 80 character limit. Line extensions can be done in python using the ```\``` character